pkg/certificate/signer.go

```diff
@@ -0,0 +1,528 @@
+		SerialNumber: serialNumber,
+		Subject: pkix.Name{
+			CommonName:   "submariner-ca",
+			Organization: []string{"submariner.io"},
+		},
+		NotBefore:             time.Now().Add(-5 * time.Minute),
+		NotAfter:              time.Now().Add(CACertValidity),
+		KeyUsage:              x509.KeyUsageCertSign | x509.KeyUsageCRLSign,
+		BasicConstraintsValid: true,
+		IsCA:                  true,
+	}
+
+	certDER, err := x509.CreateCertificate(rand.Reader, &certTemplate, &certTemplate, &privateKey.PublicKey, privateKey)
+	if err != nil {
+		return fmt.Errorf("failed to create CA certificate: %w", err)
+	}
+
+	keyPEM := pem.EncodeToMemory(&pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(privateKey)})
+	certPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certDER})
+
+	if secret.Data == nil {
+		secret.Data = map[string][]byte{}
+	}
+
+	secret.Data[CAKeyFileName] = keyPEM
+	secret.Data[CACertFileName] = certPEM
+
+	// Ensure annotations exist and set the version
+	if secret.Annotations == nil {
+		secret.Annotations = map[string]string{}
+	}
+	secret.Annotations["submariner.io/ca-version"] = currentVersion
+
+	// Create or update the secret
+	secretUnstructured := resource.MustToUnstructured(secret)
+	if secret.CreationTimestamp.IsZero() {
+		_, err = caSecretClient.Create(ctx, secretUnstructured, metav1.CreateOptions{})
+	} else {
+		_, err = caSecretClient.Update(ctx, secretUnstructured, metav1.UpdateOptions{})
+	}
+
+	if err != nil {
+		return fmt.Errorf("failed to store CA Secret: %w", err)
+	}
+
+	logger.Info("CA Secret successfully created or rotated", "namespace", namespace, "version", currentVersion)
+
+	return nil
+}
+
+// shouldReissueCA checks if the CA certificate should be reissued based on expiration time.
+// Returns (shouldReissue, newVersion, error).
+func (s *signerImpl) shouldReissueCA(secret *corev1.Secret, namespace string) (bool, string, error) {
+	// Get current version from existing secret
+	currentVersion := "1"
+	if existingVersion, exists := secret.Annotations["submariner.io/ca-version"]; exists {
+		currentVersion = existingVersion
+	}
+
+	certPEM := secret.Data[CACertFileName]
+	if len(certPEM) == 0 {
+		logger.Info("CA certificate data is empty, reissuing", "namespace", namespace)
+		return true, s.incrementVersion(currentVersion), nil
+	}
+
+	block, _ := pem.Decode(certPEM)
+	if block == nil {
+		logger.Info("Failed to decode existing CA cert PEM, reissuing", "namespace", namespace)
+		return true, s.incrementVersion(currentVersion), nil
+	}
+
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		logger.Info("Failed to parse existing CA cert, reissuing", "namespace", namespace, "error", err)
+		return true, s.incrementVersion(currentVersion), nil
+	}
+
+	timeRemaining := time.Until(cert.NotAfter)
+	logger.Info("Existing CA", "namespace", namespace, "expiresIn", timeRemaining.String(), "notAfter", cert.NotAfter.Format(time.RFC3339), "version", currentVersion)
```

Comment:
Probably TRACE here.

```suggestion
	logger.V(log.TRACE).Info("Existing CA", "namespace", namespace, "expiresIn", timeRemaining.String(), "notAfter", cert.NotAfter.Format(time.RFC3339), "version", currentVersion)
```